### PR TITLE
convert MPollBody tests into rtl

### PR DESCRIPTION
**Pull request by kerryarchibald**
Huge diff, but mostly snapshots

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

File: test/components/views/messages/MPollBody-test.tsx [hunk 1]
```diff
@@ -456,13 +470,13 @@ describe("MPollBody", () => {
             responseEvent("@catrd:example.com", "poutine"),
             responseEvent("@dune2:example.com", "wings"),
         ];
-        const body = newMPollBody(votes, [], null, false);
+        const { container } = newMPollBody(votes, [], undefined, false);
 
         // My vote is marked
-        expect(body.find('input[value="pizza"]').prop("checked")).toBeTruthy();
+        expect(container.querySelector('input[value="pizza"]')!.hasAttribute("checked")).toBeTruthy();
```
Comment (gsouquet) [inline test/components/views/messages/MPollBody-test.tsx:476]: Recently discovered that we can now do

```suggestion
        expect(container.querySelector('input[value="pizza"]')!).toHaveAttribute("checked", "true");
```

or

```suggestion
        expect(container.querySelector('input[value="pizza"]')!).toBeChecked();
```

File: test/components/views/messages/MPollBody-test.tsx [hunk 2]
```diff
@@ -577,9 +591,9 @@ describe("MPollBody", () => {
 
     it("shows non-radio buttons if the poll is ended", () => {
         const events = [endEvent()];
-        const body = newMPollBody([], events);
-        expect(body.find(".mx_StyledRadioButton")).toHaveLength(0);
-        expect(body.find('input[type="radio"]')).toHaveLength(0);
+        const { container } = newMPollBody([], events);
+        expect(container.querySelectorAll(".mx_StyledRadioButton")).toHaveLength(0);
```
Comment (gsouquet) [inline test/components/views/messages/MPollBody-test.tsx:595]: Believe we could have a more readable assertion here

```suggestion
        expect(container.querySelectorAll(".mx_StyledRadioButton")).not.toBeInTheDocument();
```

**Review by gsouquet — APPROVED** [review]
Looks good to me 👍
Added some non-blocking comments, to use some more semantic assertions... but that shouldn't block this PR from going ahead

File: test/components/views/messages/MPollBody-test.tsx (hunk 1 above)
Comment (kerryarchibald) [reply test/components/views/messages/MPollBody-test.tsx:476]: TIL! thanks

**Pull request merged**